astroplan/scheduling.py

```diff
@@ -140,6 +140,76 @@ def create_score_array(self, time_resolution=1*u.minute):
             score_array *= constraint(self.observer, targets, times)
         return score_array
 
+    def weighted_score_array(self, time_resolution=1*u.minute):
+        """
+        For each block, this returns a score for each time using the
+        formula (score*weight+score*weight+...)/(weight+weight+..)
+        """
+        # note: with this method, a non-boolean constraint with weight
+        # zero has the exact same effect if it was boolean
+        start = self.schedule.start_time
+        end = self.schedule.end_time
+        times = time_grid_from_range((start, end), time_resolution)
+        score_array = np.zeros((len(self.blocks), len(times)))
+        zeros = np.ones((len(self.blocks), len(times)), dtype=int)
```

Comment:
Ah.  Can you think of a better name than `zeros`? Maybe `zeroed_constraints` or something like that, with a comment explaining?